### ISSUE TITLE
[Feature Request] Object detection on high resolution stream

### ISSUE THREAD

**Issue by claymen** [blakeblackshear/frigate#1421]
From my understanding the current Frigate design uses the detect stream as both the motion detector and the object detection stream. To optimise for performance this means that often the lower resolution substream is used, which works well at boosting performance, however can create some problems if the camera's sub stream is not particularly high resolution.

In my setup, I have 12 camera's which offer a 1288x1520 primary stream, or a 640x360 secondary stream. Generally this performs well enough, but I have noticed that it's just not enough resolution to identify objects that are a bit further away. The easy fix would be to use the high res stream as the primary source for detection, however with 12 cameras this places a huge load on the CPU/GPU to do the video decode. A few camera's would work fine, but with all of them, it's pretty heavy.

I had looked at how some other system optimise for these situations, and noticed that another way of doing this would be to run the motion detection separate to the object detection. E.g. that the constant stream being decoded and scanned for motion detection uses the lower resolution sub stream, but when motion is detected, it runs the object detection against the higher resolution stream. In theory this could allow much lower general usage, while allowing for higher resolution/higher usage only when needed, in an on demand/motion type arrangement.

Is this something that might be possible to implement?

TLDR;
Use lower resolution substream for motion detection only, then use the higher resolution stream for object detection only when motion is detected from the low res stream.

**Comment by highaltidude** [comment]
I had a similar question. Hopefully this helps #1189

**Comment by blakeblackshear** [comment]
It's not really practical for the reasons mentioned in #1189. Frigate uses the motion detection to determine where to look in the current frame for objects. It does not simply start running object detection on the entire frame. By the time the second ffmpeg process started and passed frames, it would be several seconds too late, and it would be looking in the wrong place. You can't have a higher resolution stream ready to run object detection on without also decoding it all the time.

**Comment by claymen** [comment]
> It's not really practical for the reasons mentioned in #1189. Frigate uses the motion detection to determine where to look in the current frame for objects. It does not simply start running object detection on the entire frame. By the time the second ffmpeg process started and passed frames, it would be several seconds too late, and it would be looking in the wrong place. You can't have a higher resolution stream ready to run object detection on without also decoding it all the time.

That makes sense. Thank you for the explanation.
